extensions/lifecycle/conductor/LifecycleConductor.js

```diff
@@ -291,7 +292,11 @@ class LifecycleConductor {
     }
 
     _createBucketTaskMessages(tasks, log, cb) {
-        async.mapLimit(tasks, 10, (t, taskDone) => {
+        if (!this.lcConfig.forceLegacyListing) {
+            return process.nextTick(cb, null, tasks.map(t => this._taskToMessage(t, lifecycleTaskVersions.v1)));
+        }
+
+        return async.mapLimit(tasks, 10, (t, taskDone) => {
             if (!this.activeIndexingJobsRetrieved) {
                 return process.nextTick(taskDone, null, this._taskToMessage(t, lifecycleTaskVersions.v1));
             }
```

Comment:
I'm not sure of how the code flow behaves regarding updating `this.activeIndexingJobsRetrieved` but I think it would be possible to simplify as such:

```suggestion
        if (!this.lcConfig.forceLegacyListing || !this.activeIndexingJobsRetrieved) {
            return process.nextTick(cb, null, tasks.map(t => this._taskToMessage(t, lifecycleTaskVersions.v1)));
        }

        return async.mapLimit(tasks, 10, (t, taskDone) => {
```